docs/caching.md

```diff
@@ -0,0 +1,281 @@
+    # 👇 Added this
+    st.write("Cache miss: expensive_computation(", a, ",", b, ") ran")
+    time.sleep(2)  # This makes the function take 2s to run
+    return a * b
+
+a = 2
+b = 21
+res = expensive_computation(a, b)
+
+st.write("Result:", res)
+```
+
+Now when you rerun the app the text "Cache miss" appears on the first run, but not on any subsequent runs. That's because the cached function is only being executed once, and every time after that you're actually hitting the cache.
+
+```eval_rst
+.. note::
+   You may have noticed that we added the `suppress_st_warning` to the `@st.cache` decorators. That's because the cached function above uses a Streamlit command itself (`st.write` in this case), and when Streamlit sees that, it shows a warning that your command will only execute when you get a cache hit. More often than not, when you see that warning it's because there's a bug in your code. However, in our case we're using the `st.write` command to demonstrate when the cache is being hit, so the behavior Streamlit is warning us about is exactly what we want. As a result, we are passing in `suppress_st_warning=True` to turn that warning off.
+```
+
+## Example 2: When the function arguments change
+
+Without stopping the previous app server, let's change one of the arguments to our cached function:
+
+```python
+import streamlit as st
+import time
+
+@st.cache(suppress_st_warning=True)
+def expensive_computation(a, b):
+    st.write("Cache miss: expensive_computation(", a, ",", b, ") ran")
+    time.sleep(2)  # This makes the function take 2s to run
+    return a * b
+
+a = 2
+b = 210  # 👈 Changed this
+res = expensive_computation(a, b)
+
+st.write("Result:", res)
+```
+
+The first time you rerun the app you'll notice the text "Cache miss" and that it takes about two seconds to finish running. Now, if you press **R** to rerun the app again, it'll always be a cache hit. That means, no more text that shows "Cache miss", and you'll notice that the report loads noticeably faster.
+
+This is because Streamlit notices whenever the arguments **a** and **b** change and determines whether the function should be re-executed and re-cached.
+
+## Example 3: When the function body changes
+
+Without stopping and restarting your Streamlit server, let's remove the widget from our app and modify the function's code by adding a `+ 1` to the return value.
+
+```python
+import streamlit as st
+import time
+
+@st.cache(suppress_st_warning=True)
+def expensive_computation(a, b):
+    st.write("Cache miss: expensive_computation(", a, ",", b, ") ran")
+    time.sleep(2)  # This makes the function take 2s to run
+    return a * b + 1  # 👈 Add a +1 at the end here
+
+a = 2
+b = 210
+res = expensive_computation(a, b)
+
+st.write("Result:", res)
+```
+
+The first run is a "Cache miss", but each subsequent run is a cache hit. This is because on first run, Streamlit detected that the function body changed, reran the function, and put the result in the cache.
```

Comment:
. The first run is a "Cache miss", but when you press **R** each subsequent run is a cache hit.

Reply:
Fixed to match text above.